README.md

```diff
@@ -179,6 +179,19 @@ A Linux or MacOS environment is required. Windows is not supported and the npm c
 3. run `npm run dev` to start the webserver, open up
    http://localhost:3001/#search and try out the basic features
 
+### Pull requests
+
+Before creating a pull request:
+- commit your changes
+- update the package's version number in `package.json`
+- run `npm run sync-version`
```

Comment:
Tested this now by committing a version change to package.json only. The build succeeds, and the versions are in sync in the output produced by TC, which is certainly better than what we had before, but they remain out of sync in source control.

It would be nice to have the TC build fail in that case, so the developer is immediately informed that they forgot to sync the versions and the PR with out-of-sync versions does not get merged. Can verifyVersion.js --sync still check that the part of the version before the build number (i.e. the major+minor version) is the same?

(Alternatively, we could have that script set the build number instead of using TC's text replacement "build feature", but I think that would get more complicated than checking the first 2 parts of the version.)

Reply:
Edited my comment above because I realized the solution I was proposing was quite silly and would never work.

So maybe the version you had before (with two regexes) was the best solution, after all! Sorry about that.

Reply:
I think I finally understand where you want to go with this topic. Let me analyse our options and constraints.

The challenge we are facing is that the version verification must be a Node script that relies on placeholders (or special comments, which suck). And this script can only be run as a TC build step.
At the same time, TC must inject the build number in place of the placeholder and this can only happen in the content replacer feature which runs **before** we can run any Node script.

**Side question**: What is the purpose of the build number? Do we ever publish a minor version twice but with a different build number? E.g. `1.14.178` then `1.14.180`.

So if we want the build number, we **must** use the content replacer build feature.
If we want to catch out of sync versions in source control, we **must** use a Node script with placeholders intact.

Here is what I think would work:

`package.json` and `core.js` keep their placeholders as they are now.
We introduce a new constant with a separate placeholder in `verifyVersion.js` that gets replaced with TC's build number.
```js
const buildNumber = `TC-BUILD-NUMBER`;
```
TC only replaces this new placeholder and no longer `package` and `core`.
This way we have the build number in the code but the version placeholders are still intact.
TC triggers build step _verify version_. Here we have the state from version control. Fails if out of sync.
TC triggers _sync version_. Node script replaces placeholders in `package` and `core` with the newly introduced `buildNumber` from above.
Build.

**Locally**, we can keep the procedure of manually updating `package.json` then running _sync version_.
It's likely necessary to have separate sync scripts for local and CI build because locally, the build number is not known.
Local and CI build scripts also must likely remain separate as they currently are because of the differing placeholder availability.

Let me know your thoughts.

Reply:
> TC must inject the build number in place of the placeholder and this can only happen in the content replacer feature which runs before we can run any Node script.

I don't think it has to be there - we can inject the build number as part of a Node script, too, it was just simpler to use the built-in replacer feature for this.

But I think you ask a great question about whether the build number is needed! I don't know, but it's possible that we simply copied what we do for the engine, where we do have many builds of the same major/minor version and it's good to know the _exact_ version.

We can of course make multiple builds with the same major/minor version of the JS Connector, too, and [we have in the past](https://www.npmjs.com/package/loop54-js-connector?activeTab=versions), but I don't think we should - it's confusing to customers and inconsistent with what we do for the .NET Connector. So to prevent that it might actually be good to not have the build number in package.json (but keep it in core.js).

Then we can still do the replacement using the TC content replacer feature, as today. Extracting the build number into its own constant is still helpful, I think: then we can just verify that `"version in core.js" = "version in package.json" + buildNumber.`, so that check becomes independent of the build number.